addons/sale/models/sale_order_line.py

```diff
@@ -310,19 +315,23 @@ def _compute_no_variant_attribute_values(self):
                 if ptav._origin not in valid_values:
                     line.product_no_variant_attribute_value_ids -= ptav
 
-    @api.depends('product_id')
+    @api.depends('product_id', 'is_downpayment')
     def _compute_name(self):
         for line in self:
+            name = ''
             if not line.product_id:
-                continue
+                if line.is_downpayment and not line.display_type:
+                    name = 'Down Payment'
```

Comment:
This won't be translated correctly (and don't forget to take care of the lang of the partner)